packages/api-queries/src/me-username.ts

```diff
@@ -0,0 +1,13 @@
+import { submitUsernameChange } from '@automattic/api-core';
+import { mutationOptions } from '@tanstack/react-query';
+import { queryClient } from './query-client';
+
+export const usernameChangeMutation = () =>
+	mutationOptions( {
+		mutationFn: ( { username, action }: { username: string; action: string } ) =>
+			submitUsernameChange( username, action ),
+		onSuccess: () => {
+			// Invalidate user settings to refetch updated data
+			queryClient.invalidateQueries( { queryKey: [ 'me', 'settings' ] } );
```

Comment:
We can simply use this to avoid having to type the key


```suggestion
			queryClient.invalidateQueries( userSettingsQuery() );
```

Reply:
Updated in [4845b6b](https://github.com/Automattic/wp-calypso/pull/105912/commits/4845b6bdb9400c1ef0fad49c0f9e4460b29eae51), thank you!